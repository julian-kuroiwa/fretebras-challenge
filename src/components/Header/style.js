import styled, { css } from 'styled-components';

import Wrapper from '../Wrapper';

export const Container = styled.header`
  display: flex;
  align-items: center;
  height: 60px;
  width: 100%;
  transition: background-color 0.2s;

  ${(props) =>
    props.isScrolled &&
    css`
      background-color: #000;
      position: fixed;
      top: 0;
      left: 0;
    `}

  ul {
    list-style: none;
    display: flex;

    li {
      text-transform: uppercase;

      &:not(:last-child) {
        margin-right: 50px;
      }

      a {
        color: #fff;
        text-decoration: none;
        font-weight: bold;
      }
    }
  }
`;

export const WrapperStyled = styled(Wrapper)`
  display: flex;
  justify-content: space-between;
  align-items: center;

  > a {
    i {
      color: #fff;
      font-size: 40px;
    }
  }
`;
