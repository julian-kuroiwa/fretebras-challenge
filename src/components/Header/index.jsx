import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';

import { Container, WrapperStyled } from './style';

const Header = () => {
  const [isScrolled, setIsScrolled] = useState(false);
  const [isOpen, setIsOpen] = useState(false);

  const scrollHandle = () => {
    if (document.documentElement.scrollTop > 100) {
      setIsScrolled(true);

      return;
    }

    setIsScrolled(false);
  };

  useEffect(() => {
    window.addEventListener('scroll', scrollHandle);
    return () => {
      window.removeEventListener('scroll', scrollHandle);
    };
  });
  return (
    <Container isScrolled={isScrolled} isOpen={isOpen}>
      <WrapperStyled>
        <Link to="/">
          <i className="fas fa-hamburger" />
        </Link>

        <nav>
          <button type="button" onClick={() => setIsOpen(false)}>
            <i className="fas fa-times" />
          </button>
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/">Search</Link>
            </li>
            <li>
              <Link to="/">Promo</Link>
            </li>
            <li>
              <Link to="/">About</Link>
            </li>
            <li>
              <Link to="/">Contact</Link>
            </li>
          </ul>
        </nav>

        <button type="button" onClick={() => setIsOpen(true)}>
          <i className="fas fa-bars" />
        </button>
      </WrapperStyled>
    </Container>
  );
};

export default Header;
