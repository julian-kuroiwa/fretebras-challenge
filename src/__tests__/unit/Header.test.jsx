import { shallow } from 'enzyme';
import React from 'react';

import Header from '../../components/Header';

describe('Header', () => {
  it('should match snapshot', () => {
    const component = shallow(<Header />);

    expect(component).toMatchSnapshot();
  });
});
