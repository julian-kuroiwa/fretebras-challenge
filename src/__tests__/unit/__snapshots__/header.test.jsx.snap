// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<styled.header
  isScrolled={false}
>
  <Styled(Wrapper)>
    <a
      href="#"
    >
      <i
        className="fas fa-hamburger"
      />
    </a>
    <nav>
      <ul>
        <li>
          <a
            href="#"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Search
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Promo
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
  </Styled(Wrapper)>
</styled.header>
`;
